sufia-models/app/actors/sufia/generic_file/actor.rb

```diff
@@ -46,7 +46,12 @@ def add_file_to_collection(collection_id)
       return unless user.can? :edit, collection
       acquire_lock_for(collection_id) do
         collection.add_members [generic_file.id]
-        collection.save
+        begin
+          collection.save
+        rescue StandardError => error
+          Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::add_file_to_collection Error adding #{generic_file.title} to collection #{collection.title}. Caught Exception error #{error.inspect}"
```

Comment:
There is no Rails magic.  Sufic.config.upload_logger is an instance of Logger that is created in initializers/sufia.rb.